### PR TITLE
Services and repositories

### PR DESCRIPTION
**Pull request by AlexandrPotovichenko**
Add Services and Repositories

### CODE REVIEW

File: ProjectManagement.BusinessLogic/Exceptions/GuardExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+﻿
+//using Microsoft.eShopWeb.ApplicationCore.Exceptions;
+using ProjectManagement.BusinessLogic.Exceptions;
+using ProjectManagement.Domain.Models;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Ardalis.GuardClauses
```
Comment (mykolazaiarnyi) [inline ProjectManagement.BusinessLogic/Exceptions/GuardExtensions.cs:8]: Please, name your namespaces the same as a path in a file system. In this case it should be `ProjectManagement.BusinessLogic.Exceptions`

File: ProjectManagement.BusinessLogic/Exceptions/GuardExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+{
+    public static class BoardGuards
```
Comment (mykolazaiarnyi) [inline ProjectManagement.BusinessLogic/Exceptions/GuardExtensions.cs:10]: Class name should be the same as a file name

File: ProjectManagement.BusinessLogic/Services/Implementation/BoardService.cs [hunk 3]
```diff
@@ -1,70 +1,108 @@
...
+using System.Linq;
+using System;
 
 namespace ProjectManagement.BusinessLogic.Services.Implementation
 {
     public class BoardService : IBoardService
     {
         private readonly IBoardRepository _BoardRepository;
         private readonly IUserRepository _UserRepository;
-        //private readonly IBoardMemberRepository _BoardMemberRepository;
```
Comment (mykolazaiarnyi) [inline ProjectManagement.BusinessLogic/Services/Implementation/BoardService.cs:15]: It's considered a bad practice to keep commented code

File: ProjectManagement.BusinessLogic/Services/Interfaces/IBoardService.cs [hunk 4]
```diff
@@ -6,8 +6,17 @@ namespace ProjectManagement.BusinessLogic.Services.Interfaces
...
     public interface IBoardService
     {
+
+       
+        Task<Board> CreateBoardAsync(string name, string description, int userId);
+        Task<Board> GetBoardAsync(int boardId);
         Task<IEnumerable<Board>> GetBoardsAsync();
-        Task<Board> CreateBoardAsync(string name, 
-            string description, int userId);
+        Task DeleteBoard(int boardId);
```
Comment (mykolazaiarnyi) [inline ProjectManagement.BusinessLogic/Services/Interfaces/IBoardService.cs:14]: Please, add the `async` suffix to the name of async methods

**Review by mykolazaiarnyi — CHANGES_REQUESTED** [review]

**Comment by AlexandrPotovichenko** [comment]
Fixed bugs.

**Pull request merged**